Donation-platform/src/main/java/com/google/PersonalizedRecommendations.java

```diff
@@ -50,18 +54,22 @@
   private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_1_AND_3 = 0.15;
   private final double TAGSCORE_INCREMENT_FOR_HAVING_TAG_2_AND_3 = 0.1;
 
-  // Constructor to setup db (and populate db if necessary)
+  // Constructor to setup db
   public PersonalizedRecommendations(DatastoreService ds) {
     this.ds = ds;
     db = new DbCalls(ds);
-    DbSetUpUtils setup = new DbSetUpUtils(ds, db);
-    // Call FindTrendingCharities so that trending scores can be considered in personalized order
-    FindTrendingCharities findTrending = new FindTrendingCharities(ds);
-    findTrending.queryDb();
+    AddCharitiesFromJSON setup = new AddCharitiesFromJSON(ds, db);
```

Comment:
why does this need ds? db is already initialized and you don't do anything with it in the class either

Reply:
AddCharitiesFromJSON needs to use the same datastore for populating charities and tags if necessary. The class cannot initialize ds and db without those being passed in because it can't it just start adding charities/tags to another datastore (@tonioshikanlu let me know if wrong on this).